front/src/reducers/infra/index.ts

```diff
@@ -0,0 +1,44 @@
+import { createSlice } from '@reduxjs/toolkit';
+import type { CaseReducer, Draft, PayloadAction } from '@reduxjs/toolkit';
+import type { Feature } from 'geojson';
+
+export interface InfraStateType {
+  infraID: number | undefined;
+  featureInfoClick: { displayPopup: boolean; feature?: Feature; coordinates?: number[] };
+}
+export const infraState: InfraStateType = {
+  infraID: undefined,
+  featureInfoClick: { displayPopup: false },
+};
+
+type InfraStateReducersType<S extends InfraStateType> = {
+  ['updateInfraID']: CaseReducer<S, PayloadAction<S['infraID']>>;
+  ['updateFeatureInfoClick']: CaseReducer<S, PayloadAction<S['featureInfoClick']>>;
+};
+
+export function buildInfraStateReducers<S extends InfraStateType>(): InfraStateReducersType<S> {
+  return {
+    updateInfraID(state: Draft<S>, action: PayloadAction<S['infraID']>) {
+      state.infraID = action.payload;
+    },
+    updateFeatureInfoClick(state: Draft<S>, action: PayloadAction<S['featureInfoClick']>) {
+      state.featureInfoClick = action.payload;
+    },
+  };
+}
+
+export const infraSlice = createSlice({
+  name: 'infra',
+  initialState: infraState,
+  reducers: {
+    ...buildInfraStateReducers<InfraStateType>(),
+  },
+});
+
+export const infraSliceActions = infraSlice.actions;
+
+export type infraSliceType = typeof infraSlice;
+
+export type infraSliceActionsType = typeof infraSliceActions;
+
+export default infraSlice.reducer;
```

Comment:
From what I can, all this code is never used, so I think we can remove it
```suggestion
```

Reply:
Done